sematic/plugins/building/docker_builder.py

```diff
@@ -139,6 +180,7 @@ class BuildConfig:
     """
 
     version: int
+    project: str
```

Comment:
There was a bug where if the `base_uri` of the base image was used locally to build the local image (by adding local files, requirements, and entry point) under the same name (effectively `sematicai/sematic-worker-base:latest@sha256:...`), it would actually just overwrite the `latest` tag, and would create the situation in which the user had a `sematicai/sematic-worker-base:latest` locally that actually held their project's image.

For us, or for users using custom base images pointing to their repos, this would create the risk of confusion, and in extreme cases even the risk of overwriting the base image with project-specific images if a push were to be made.

So I introduced this field, which already exists in the Bazel build system, and am using it to generate a local name for the project build image.

Reply:
Instead of adding a new configuration parameter for users, can we just use an the name of the directory the config file is in as the "project name"? So building with the config file:

`foo/bar/main.yaml` would yield something like `bar:default@sha256...`

We should aim to keep the cognitive overhead of configuration as minimal as possible and try to avoid introducing new concepts like "project". Also we might need some "project" concept in the future that conflicts with this usage.

If we *must* add a new config parameter, perhaps we could name it `local_image_name` to make it more clear and less likely to create a concept-clash later?

Reply:
Great suggestion! I want to keep this naming convention forward-compatible with future multi-base image support. So how about:

```yaml
version: 1
images:
  - example_docker:
      base_uri: "sematicai/sematic-worker-base:latest@sha256:bea3926876a3024c33fe08e0a6b2c0377a7eb600d7b3061a3f3f39d711152e3c"
      tag_suffix: "example_docker_intermediate"
  - example_docker_cuda:
      build_script: <my build script>
build:
  platform: "linux/amd64"
  requirements: "requirements.txt"
  data: ["intermediate/data/message.txt"]
  src: ["intermediate/**.py"]
push:
  registry: <image push registry>
  repository: <image push repository>
  tag_suffix: <optional image push tag suffix>
```
cc @neutralino1

Reply:
Thought about this some. Initially I wanted to keep the naming scheme forward-compatible with multiple base images, but I don't see any way of doing that and also keeping the single image version from suggesting multiple base image support would exist.

Also, it would not be symmetrical with the Bazel build system, which does not have this problem at all, as it does not expose the local images in the Docker image list. Also, the Bazel version does not have the option pf specifying a build script, which is expected to provide an image name anyway, but not required to provide a local image name which can be overridden.

So I'm going with your suggestion of using the directory name as the local name.